Lib/test/test_ast/test_ast.py

```diff
@@ -1614,6 +1614,17 @@ def test_literal_eval_syntax_errors(self):
                 (\
             \ ''')
 
+    def test_literal_eval_large_input_crash(self):
+        # gh-125010: Fix use-after-free in ast repr()
+        # Note: Without setting sys.set_int_max_str_digits(0),
+        # this code throws a 'ValueError: Exceeds the limit (4300 digits)'.
+        # With sys.set_int_max_str_digits(0),
+        # this code throws a 'ValueError: malformed node or string on line 1':
+        source = "{0x0" + "e" * 250_000 + "%" + "e" * 250_000 + "1j}"
+        with self.assertRaisesRegex(ValueError,
+                                    r"Exceeds the limit \(4300 digits\)"):
```

Comment:
I'm a bit concerned about using `assertRaisesRegex` here. Especially the part `4300 digits`. The chance of this number changing in the future is relatively small, but it doesn't equal zero :-)

I think that `assertRaises(ValueError)` would be enough here because there are comments above those statements explaining what kind of `ValueError` can be raised here.

Otherwise LGTM.

Reply:
```suggestion
        with self.assertRaisesRegex(ValueError,
                                    r"Exceeds the limit \(\d+ digits\)"):
```
Maybe just regex out the exact number. It's good to test the error message to make sure we raise a ValueError that is related to what we're testing.

Reply:
Added ;)